pynicotine/gtkgui/userbrowse.py

```diff
@@ -361,8 +421,22 @@ def on_expand(self, widget):
             self.expand.set_property("icon-name", "go-down-symbolic")
 
     def on_folder_row_activated(self, treeview, path, column):
-        if self.user != config.sections["server"]["login"]:
-            self.on_download_directory()
+
+        if path is None:
+            return
+
+        # Keyboard accessibility support for <Return> key behaviour
+        if self.FolderTreeView.row_expanded(path):
+            expandable = self.FolderTreeView.collapse_row(path)
+        else:
+            expandable = self.FolderTreeView.expand_row(path, False)
+
+        if not expandable and len(self.file_store) > 0:
+            # This is the deepest level, so move focus over to Files if there are any
+            self.FileTreeView.grab_focus()
+
+        # Note: Other Folder actions are handled by setup_accelerator functions [Shift/Ctrl/Alt+Return]
+        # ToDo: Mouse double-click actions will need *args for keycode state & mods [Ctrl/Alt+DblClick]
```

Comment:
What does this mean?

Reply:
My intention is to trigger some useful accelerator functions from modified mouse click events, but I can't implement this here because the state of the `Ctrl`, `Alt` and `Shift` keys are not easily available as arguments.

For example `Ctrl+DblClick` = (the same as Ctrl+Return) -> `on_folder_transfer_to_accelerator()`

Perhaps a new or existing function in utils.py could serve this purpose?

Reply:
Right, a whole different mess to deal with. Might have to look into this and "row-activated" when rewriting treeviews in a future Nicotine+ version.

Reply:
Indeed. I tried looking into it and it got hacky very quickly, a more robust solution would be desirable. Maybe it is not only TreeViews but also widgets like TextEntrys and buttons could benefit with some Ctrl actions or Shift overrides, maybe even some menu-items might benefit from the flexibility offered by an application wide solution.

Reply:
It also needs to work in GTK4, so most of the existing solutions online are probably invalid at this point. A fun afternoon/weekend project in any case.